tests/test_managed_agents.py

```
@@ -0,0 +1,737 @@
+    0.5 and 5.0 are refused too. Sorting representable floats from lossy ones
+    is the trap — a caller passing floats eventually passes a lossy one.
+    """
+    with pytest.raises(TypeError) as exc:
+        to_money_string(amount)
+    assert "cannot be a float" in str(exc.value)
+
+
+def test_the_refusal_does_not_hand_back_the_caller_s_own_rounding_error():
+    """repr(0.1 + 0.2) is "0.30000000000000004". Quoting it as the suggested
+    fix would recommend exactly the value the refusal exists to prevent."""
+    message = str(pytest.raises(TypeError, to_money_string, 0.1 + 0.2).value)
```

Ruff (0.16.1)

[warning] Use context-manager form of `pytest.raises()`

(RUF061)